### ISSUE TITLE
Is possible to build and run eshop for testing the event bus on local machine?

### ISSUE THREAD

**Issue by thikon** [dotnet-architecture/eShopOnContainers#678]
![image](https://user-images.githubusercontent.com/347717/43133899-5ee065ae-8f6a-11e8-905b-3d93e260e0cd.png)

Ref : [https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events](url)

If the catalog api published a message to the azure service bus ,how can we check what the message on subscribe (basket api) ? or we need to integrate the webhook or else?

Thanks,

**Comment by CESARDELATORRE** [comment]
If you want to test it using the local machine, then the Event Bus implementation is what we have by default which is using a RabbitMQ container deployed locally.
You can do a test:
1. Add an article to the basket using the web app
2.  With Swagger UI, access directly the Catalog microservice (http://localhost:5101), change the price of the same article you added to the basket, with the PUT method, and an integration event will be sent so other microservices are aware, like the basket.
3. Enter into the basket and see the business message saying that the price was changed,

Azure Service Bus is optional and you'd use it when "moving to production" to a scalable solution in Azure.
However, you can also run all the application containers locally except the Event Bus on Azure Service Bus running in Azure. That's also possible.
Closing the issue. Open if you have further questions.

**Comment by thikon** [comment]
How the receiver know the new message from the sender? or we need to find the 3rd party for callback?

![image](https://user-images.githubusercontent.com/347717/43186852-272ac0e0-901a-11e8-988f-eb83f60244b7.png)

we created a new project and implement like these and we got a problem the sender could sent a message to the Event Bus but the receiver could not get some message, so how should we do?

Thanks,
